ddtrace/settings.py

```diff
@@ -102,18 +122,80 @@ def __init__(self, global_config, *args, **kwargs):
         :param kwargs:
         """
         super(IntegrationConfig, self).__init__(*args, **kwargs)
-
+        self.http = HttpConfig()
         self.global_config = global_config
 
     def __deepcopy__(self, memodict=None):
         new = IntegrationConfig(self.global_config, deepcopy(dict(self)))
+        new.http = deepcopy(self.http)
         return new
 
     def __repr__(self):
         cls = self.__class__
         keys = ', '.join(self.keys())
         return '{}.{}({})'.format(cls.__module__, cls.__name__, keys)
 
+    def header_is_traced(self, header_name):
+        """
+        Returns whether or not the current header should be traced.
+        :param header_name: the header name
+        :type header_name: str
+        :rtype: bool
+        """
+        return self.http.header_is_traced(header_name) \
```

Comment:
Please do not use `\` to breakup lines.

Instead this could be:

```python
return (
    self.http.header_is_traced(header_name)
    if self.http.is_header_tracing_configured
    else self.global_config.header_is_traced(header_name)
)
```